dubbo-common/src/main/java/org/apache/dubbo/common/io/UnsafeStringWriter.java

```diff
@@ -80,7 +84,8 @@ public Writer append(CharSequence csq) {
     }
 
     @Override
-    public Writer append(CharSequence csq, int start, int end) {
+    @SuppressWarnings("argument.type.incompatible") // The documentation is inherited from the overridden method.
```

Comment:
I don't understand what you mean by this? Are you saying that the spec on the inherited method is wrong?

Reply:
I tried to say that the implementation of this method is similar to the overridden one, which is part of the JDK. That one has a documentation and specifies that the method can throw an `IndexOutOfBoundsException`

Reply:
Can you clarify in the comment exactly where the warning is being suppressed, and how the exception can be thrown? Alternatively, you could add a link to the JDK documentation on the web so that a reader can easily read it themselves. As it is now, it's very confusing.